packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/useChartCartesianInteraction.selectors.ts

```diff
@@ -0,0 +1,155 @@
+import { createSelector } from '../../utils/selectors';
+import { AxisId, ChartsAxisProps } from '../../../../models/axis';
+import {
+  selectorChartsInteractionPointerX,
+  selectorChartsInteractionPointerY,
+} from '../useChartInteraction/useChartInteraction.selectors';
+import { getAxisIndex, getAxisValue } from './getAxisValue';
+import { selectorChartXAxis, selectorChartYAxis } from './useChartCartesianAxisRendering.selectors';
+import { ComputeResult } from './computeAxisValue';
+
+const optionalGetAxisId = (_: unknown, id?: AxisId) => id;
+const optionalGetAxisIds = (_: unknown, ids: AxisId[]) => ids;
+
+/**
+ * Get interaction indexes
+ */
+
+function indexGetter(value: number, axes: ComputeResult<ChartsAxisProps>, ids?: AxisId): number;
+function indexGetter(value: number, axes: ComputeResult<ChartsAxisProps>, ids: AxisId[]): number[];
+function indexGetter(
+  value: number,
+  axes: ComputeResult<ChartsAxisProps>,
+  ids: AxisId | AxisId[] = axes.axisIds[0],
+): number | number[] {
+  return Array.isArray(ids)
+    ? ids.map((id) => getAxisIndex(axes.axis[id], value))
+    : getAxisIndex(axes.axis[ids], value);
+}
+
+export const selectorChartsInteractionXAxisIndex = createSelector(
+  [selectorChartsInteractionPointerX, selectorChartXAxis, optionalGetAxisId],
+  (value, axes, id) => (value === null ? null : indexGetter(value, axes, id)),
+);
+
+export const selectorChartsInteractionXAxisIndexes = createSelector(
+  [selectorChartsInteractionPointerX, selectorChartXAxis, optionalGetAxisIds],
+  (value, axes, ids) => (value === null ? null : indexGetter(value, axes, ids)),
+);
+
+export const selectorChartsInteractionYAxisIndex = createSelector(
+  [selectorChartsInteractionPointerY, selectorChartYAxis, optionalGetAxisId],
+  (value, axes, id) => (value === null ? null : indexGetter(value, axes, id)),
+);
+
+export const selectorChartsInteractionYAxisIndexes = createSelector(
+  [selectorChartsInteractionPointerY, selectorChartYAxis, optionalGetAxisIds],
+  (value, axes, ids) => (value === null ? null : indexGetter(value, axes, ids)),
+);
+
+/**
+ * Get interaction values
+ */
+
+type Value = number | Date | null;
+
+function valueGetter(
+  value: number,
+  axes: ComputeResult<ChartsAxisProps>,
+  indexes: number,
+  ids?: AxisId,
+): Value;
+function valueGetter(
+  value: number,
+  axes: ComputeResult<ChartsAxisProps>,
+  indexes: number[],
+  ids: AxisId[],
+): Value[];
+function valueGetter(
+  value: number,
+  axes: ComputeResult<ChartsAxisProps>,
+  indexes: number | number[],
+  ids: AxisId | AxisId[] = axes.axisIds[0],
+): Value | Value[] {
+  return Array.isArray(ids)
+    ? ids.map((id, axisIndex) =>
+        getAxisValue(axes.axis[id], value, (indexes as number[])[axisIndex]),
+      )
+    : getAxisValue(axes.axis[ids], value, indexes as number);
+}
```

Comment:
Would it be easier to always accept an array and always return an array as well? We wouldn't need these overloads otherwise